feg/gateway/services/n7_proxy/servicers/session_controller.go

```diff
@@ -0,0 +1,211 @@
+	"magma/feg/gateway/services/n7_proxy/metrics"
+	"magma/lte/cloud/go/protos"
+	orcprotos "magma/orc8r/lib/go/protos"
+)
+
+type CentralSessionController struct {
+	policyClient  *n7_client.ClientWithResponses
+	dbClient      policydb.PolicyDBClient
+	cfg           *SessionControllerConfig
+	healthTracker *metrics.SessionHealthTracker
+}
+
+type SessionControllerConfig struct {
+	PCFConfig      *PCFConfig
+	RequestTimeout time.Duration
+	DisableN7      bool
+}
+
+func newSessionControllerConfig(config *N7ProxyConfig) *SessionControllerConfig {
+	return &SessionControllerConfig{
+		PCFConfig:      config.Server,
+		RequestTimeout: 3 * time.Second,
+		DisableN7:      config.DisableN7,
+	}
+}
+
+func NewCentralSessionController(
+	config *N7ProxyConfig,
+	dbClient policydb.PolicyDBClient,
+) (*CentralSessionController, error) {
+	policyClient, err := createN7Client(config.Server)
+	if err != nil {
+		glog.Errorf("Creating N7 Client failed: %s", err)
+		return nil, err
+	}
+	return &CentralSessionController{
+		policyClient:  policyClient,
+		dbClient:      dbClient,
+		cfg:           newSessionControllerConfig(config),
+		healthTracker: metrics.NewSessionHealthTracker(),
+	}, nil
+}
+
+// CreateSession begins a UE session by requesting rules from PCF and returning them.
+func (srv *CentralSessionController) CreateSession(
+	ctx context.Context,
+	request *protos.CreateSessionRequest,
+) (*protos.CreateSessionResponse, error) {
+
+	return nil, nil
+}
+
+// UpdateSession handles periodic updates from gateways that include quota
+// exhaustion and terminations.
+func (srv *CentralSessionController) UpdateSession(
+	ctx context.Context,
+	request *protos.UpdateSessionRequest,
+) (*protos.UpdateSessionResponse, error) {
+
+	return nil, nil
+}
+
+// TerminateSession handles a session termination
+func (srv *CentralSessionController) TerminateSession(
+	ctx context.Context,
+	request *protos.SessionTerminateRequest,
+) (*protos.SessionTerminateResponse, error) {
+
+	return nil, nil
+}
+
+// createN7Client creates a N7 oapi client and sets the OAuth2 client credentiatls for authorizing requests
+func createN7Client(cfg *PCFConfig) (*n7_client.ClientWithResponses, error) {
+	httpClient := &http.Client{}
+	tokenConfig := clientcredentials.Config{
+		ClientID:     cfg.ClientId,
+		ClientSecret: cfg.ClientSecret,
+		TokenURL:     cfg.TokenUrl,
+	}
+	tokenCtxt := context.WithValue(context.Background(), oauth2.HTTPClient, httpClient)
+	doer := tokenConfig.Client(tokenCtxt)
+
+	serverStr, err := GetServerStringFromUrl(cfg.ApiRoot)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create new N7 client object and assosiate it with oAuth2 HTTP client
+	client, err := n7_client.NewClientWithResponses(
+		serverStr,
+		n7_client.WithHTTPClient(doer),
+	)
+
+	return client, err
+}
+
+// Disable closes all existing pcf connections and disables
+// connection creation for the time specified in the request
+func (srv *CentralSessionController) Disable(
```

Comment:
If I recall correctly health service is related with High Availability (HA) feature. When health is bad, orc8r will switch to another FEG and then send disable message (the other feg will receive enable)

However, that enable/disable was needed for diameter (its connection is stateful, so you need to disconnect from the server and release the client). But  per what I understand 5g is connectionless which means there client doesn't really need to be enabled.

If this is the case you can just implement thos functions to do nothing